burn-slashed-collateral.md

```diff
@@ -0,0 +1,123 @@
+<pre>
+  WIP: WIP-0023
+  Layer: Consensus (hard fork)
+  Title: Burn slashed collateral
+  Authors: drcpu <drcpu@protonmail.com>
+  Discussions-To: https://github.com/witnet/witnet-rust/discussions/2238
+  Status: Draft
+  Type: Standards Track
+  Created: ????-??-??
+  License: BSD-2-Clause
+</pre>
+
+## Abstract
+
+This improvement proposal introduces the idea of burning slashed collateral to prevent malicious node operators from profiting of collateral-stealing attacks.
+
+## Motivation and rationale
+
+Succesfully solving data requests happens in four different stages. First the data request is selected from the memory pool and included in a block. Now it moves to the commitment stage where nodes calculate whether they are eligible to solve it. When they are eligibile, they will solve the data request by quering all requested API's. After enough commit transactions have been broadcasted, the data request moves to the reveal stage. Here the nodes that broadcasted a commitment transaction will broadcast the associated reveal. After the network has either seen enough reveals, the data request will move to the tally stage. Here all reveals will be analyzed following the rules defined by the data request. Nodes which reveal an in-consensus value are rewarded, nodes which reveal an error receive their collateral back without a reward and nodes which reveal an out-of-consensus value are punished by the network confiscating the collateral they put up. This proposal focuses on the nodes that reveal an out-of-consensus variable.
+
+Confiscated collateral is currently redistributed to the nodes that solved a data request honestly. The problem with that approach is that it creates an obvious incentive for malicious nodes operators to perform several types of collateral-stealing attacks on other witnesses. Two types of attacks are described in [the discussion](burn-slashed-collateral) that lead to this WIP.
+
+The first and easiest to execute attack revolves around creating a data request with 'inconsistent' sources. This goal here is for a number of witnesses to reveal an out-of-consensus value such that the quorum requirements are still satisifed. This results in a number of coins (`liars * collateral`) being slashed and redistributed amongst honest nodes. The attacker does not directly benefit from creating this specific data request because he has no control over which nodes earn the redistributed collateral. However, if he repeats this attack often enough and controls a non-negligible part of the ARS his nodes will statistically gain more Witnet coins than if the attack was not taking place due to his nodes receiving part of the slashed coins.
```

Comment:
```suggestion
The first and easiest to execute attack revolves around creating a data request with 'inconsistent' sources. This goal here is for a number of witnesses to reveal an out-of-consensus value such that the quorum requirements are still satisfied. This results in a number of coins (`liars * collateral`) being slashed and redistributed amongst honest nodes. The attacker does not directly benefit from creating this specific data request because he has no control over which nodes earn the redistributed collateral. However, if he repeats this attack often enough, and controls a non-negligible part of the ARS, his nodes will statistically gain more Witnet coins than if the attack was not taking place due to his nodes receiving part of the slashed coins.
```